From: Greg Rose <gregory.v.rose@intel.com>
Date: Fri, 22 Jul 2011 05:46:07 +0000
Subject: pci: Add flag indicating device has been assigned by KVM
Patch-mainline: v3.2-rc1
Git-commit: 6777829cfe1c4ed78319ad40aaee60254222da76
References: bnc#777565 FATE#313819

Device drivers that create and destroy SR-IOV virtual functions via
calls to pci_enable_sriov() and pci_disable_sriov can cause catastrophic
failures if they attempt to destroy VFs while they are assigned to
guest virtual machines.  By adding a flag for use by the KVM module
to indicate that a device is assigned a device driver can check that
flag and avoid destroying VFs while they are assigned and avoid system
failures.

CC: Ian Campbell <ijc@hellion.org.uk>
CC: Konrad Wilk <konrad.wilk@oracle.com>
Signed-off-by: Greg Rose <gregory.v.rose@intel.com>
Acked-by: Jesse Barnes <jbarnes@virtuousgeek.org>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
Automatically created from "patches.fixes/pci-Add-flag-indicating-device-has-been-assigned-by-.patch" by xen-port-patches.py

--- a/drivers/xen/pciback/pci_stub.c
+++ b/drivers/xen/pciback/pci_stub.c
@@ -113,6 +113,7 @@ static void pcistub_device_release(struc
 	kfree(pci_get_drvdata(dev));
 	pci_set_drvdata(dev, NULL);
 
+	dev->dev_flags &= ~PCI_DEV_FLAGS_ASSIGNED;
 	pci_dev_put(dev);
 
 	kfree(psdev);
@@ -351,6 +352,7 @@ static int __devinit pcistub_init_device
 	dev_dbg(&dev->dev, "reset device\n");
 	pciback_reset_device(dev);
 
+	dev->dev_flags |= PCI_DEV_FLAGS_ASSIGNED;
 	return 0;
 
       config_release:
